src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala

```diff
@@ -0,0 +1,183 @@
+package temple.generate.server.go.service.main
+
+import temple.ast.AttributeType.DateTimeType
+import temple.ast.{Annotation, AttributeType}
+import temple.generate.CRUD._
+import temple.generate.server.{CreatedByAttribute, ServiceRoot}
+import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.generate.server.go.common.GoCommonGenerator._
+import temple.utils.StringUtils.doubleQuote
+
+import scala.Option.when
+import scala.collection.immutable.ListMap
+
+object GoServiceMainHandlersGenerator {
+
+  /** Generate a map for converting the fields of the DAO response to the JSON response  */
+  private def generateResponseMap(root: ServiceRoot): ListMap[String, String] =
```

Comment:
This function feels like it's a bit too much to be one expression, how about breaking it down?